motionplan/rrt.go

```diff
@@ -132,29 +136,55 @@ func (n *basicNode) Q() []referenceframe.Input {
 	return n.q
 }
 
+func (n *basicNode) Cost() float64 {
+	return math.NaN()
+}
+
+func (n *basicNode) Pose() spatialmath.Pose {
+	return nil
+}
+
 type costNode struct {
 	node
 	cost float64
 }
 
-func newCostNode(q []referenceframe.Input, cost float64) *costNode {
+func newCostNode(q []referenceframe.Input, cost float64) node {
 	return &costNode{&basicNode{q: q}, cost}
 }
 
+func (n *costNode) Cost() float64 {
```

Comment:
At this point is there even a reason to have the different types of nodes and the node interface in general?  The way this is written its effectively the same as everything being the new Configuration node that you have introduced, except nodes with less functionality get implicitly passed NaN and nil into cost and pose respectively.  If we're going to go this way I feel like we might as well simplify a little.  Thoughts?

Reply:
Yeah that's probably correct.

Reply:
I want to leave it an interface to allow for the possibility of more complex behavior but am good to make everything just one data structure now